cmd/clusterctl/pkg/client/cluster/mover.go

```diff
@@ -84,6 +90,54 @@ func newObjectMover(fromProxy Proxy, log logr.Logger) *objectMover {
 	}
 }
 
+// checkProvisioningCompleted checks if Cluster API has already completed the provisioning of the infrastructure for the objects involved in the move operation.
+func (o *objectMover) checkProvisioningCompleted(graph *objectGraph) error {
+	errList := []error{}
+	cFrom, err := o.fromProxy.NewClient()
+	if err != nil {
+		return err
+	}
+
+	// Checking all the clusters have infrastructure is ready
+	for _, cluster := range graph.getClusters() {
+		clusterObj := &clusterv1.Cluster{}
+		clusterObjKey := client.ObjectKey{
+			Namespace: cluster.identity.Namespace,
+			Name:      cluster.identity.Name,
+		}
+
+		if err := cFrom.Get(ctx, clusterObjKey, clusterObj); err != nil {
+			return errors.Wrapf(err, "error reading %q %s/%s",
+				clusterObj.GroupVersionKind(), clusterObj.GetNamespace(), clusterObj.GetName())
+		}
+
+		if !clusterObj.Status.InfrastructureReady {
+			errList = append(errList, errors.Errorf("cannot start the move operation while  %q %s/%s is still provisioning the infrastructure", clusterObj.GroupVersionKind(), clusterObj.GetNamespace(), clusterObj.GetName()))
```

Comment:
Do we want to return over here? That is, should we be checking the machine node-refs if `clusterObj.Status.InfrastructureReady` is false. 
Suggesting a return here because it seems like an expensive operation to get the machines.

Reply:
As a user, I would like to get all the errors in one run; 

>  That is, should we be checking the machine node-refs if clusterObj.Status.InfrastructureReady is false

There can be more than one cluster, so we can't return if only one has InfrastructureReady false.

Reply:
Ah makes sense. I just read the docs for what the `clusterctl move` command was supposed to do.